sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/ByteBuddyOnTimerInvokerFactory.java

```diff
@@ -203,12 +205,26 @@ public String subclass(TypeDescription.Generic superClass) {
 
     private final DoFnSignature.OnTimerMethod signature;
 
-    public InvokeOnTimerDelegation(DoFnSignature.OnTimerMethod signature) {
-      super(signature.targetMethod());
+    public InvokeOnTimerDelegation(
+        TypeDescription clazzDescription, DoFnSignature.OnTimerMethod signature) {
+      super(clazzDescription, signature.targetMethod());
       this.signature = signature;
     }
 
     @Override
+    public InstrumentedType prepare(InstrumentedType instrumentedType) {
+      // Remember the field description of the instrumented type.
+      // Kind of a hack to set the protected value
```

Comment:
May be worth noting that this is done because instrumentedType is only available to prepare, but it is needed in beforeDelegation. It just so happens that we know prepare is called before beforeDelegation (I think)